lib/plugins/aws/deploy/lib/monitorStack.js

```diff
@@ -0,0 +1,113 @@
+
+module.exports = {
+  monitorStack(action, cfData, frequency) {
+    // Skip monitoring if a deployment should not be performed
+    if (this.options.noDeploy) return BbPromise.bind(this).then(BbPromise.resolve());
+
+    // Skip monitoring if stack was already created
+    if (typeof cfData === 'undefined') return BbPromise.bind(this).then(BbPromise.resolve());
+
+    // Monitor stack creation/update/removal
+    const validStatuses = [
+      'CREATE_COMPLETE',
+      'UPDATE_COMPLETE',
+      'DELETE_COMPLETE',
+    ];
+    const invalidStatuses = [
+      'CREATE_FAILED',
+      'DELETE_FAILED',
+      'ROLLBACK_FAILED',
+      'UPDATE_ROLLBACK_COMPLETE',
+      'UPDATE_ROLLBACK_FAILED',
+    ];
+    const loggedEvents = [];
+    const monitoredSince = new Date();
+    monitoredSince.setSeconds(monitoredSince.getSeconds() - 5);
+
+    let stackStatus = null;
+    let stackLatestError = null;
+
+    this.serverless.cli.log(`Checking Stack ${action} progress...`);
+
+    return new BbPromise((resolve, reject) => {
+      async.whilst(
+        () => (validStatuses.indexOf(stackStatus) === -1),
+        (callback) => {
+          setTimeout(() => {
+            const params = {
+              StackName: cfData.StackId,
+            };
+            return this.sdk.request('CloudFormation',
+              'describeStackEvents',
+              params,
+              this.options.stage,
+              this.options.region)
+              .then((data) => {
+                // Loop through stack events
+                data.StackEvents.reverse().forEach((event) => {
+                  const eventInRange = (monitoredSince < event.Timestamp);
+                  const eventNotLogged = (loggedEvents.indexOf(event.EventId) === -1);
+                  let eventStatus = event.ResourceStatus;
+                  if (eventInRange && eventNotLogged) {
+                    // Keep track of stack status
+                    if (event.ResourceType === 'AWS::CloudFormation::Stack') {
+                      stackStatus = eventStatus;
+                    }
+                    // Keep track of first failed event
+                    if (eventStatus.endsWith('FAILED') && stackLatestError === null) {
+                      stackLatestError = event;
+                    }
+                    // Log stack events
+                    if (this.options.verbose) {
+                      if (eventStatus.endsWith('FAILED')) {
+                        eventStatus = chalk.red(eventStatus);
+                      } else if (eventStatus.endsWith('PROGRESS')) {
+                        eventStatus = chalk.yellow(eventStatus);
+                      } else if (eventStatus.endsWith('COMPLETE')) {
+                        eventStatus = chalk.green(eventStatus);
+                      }
+                      /* eslint-disable max-len */
+                      this.serverless.cli.consoleLog(`CloudFormation - ${eventStatus} - ${event.ResourceType} - ${event.LogicalResourceId}`);
```

Comment:
Could be split up e.g. like this so that we can remove the linting disabling:

```
this.serverless.cli.consoleLog(`CloudFormation - ${
eventStatus} - ${
event.ResourceType} - ${
event.LogicalResourceId}`);
```